lockfile/src/parsers/pypi.rs

```diff
@@ -85,7 +98,13 @@ fn package(input: &str) -> IResult<&str, Package> {
 
     // Parse first-party dependencies.
     let (input, version) = package_version(input)?;
-    let version = PackageVersion::FirstParty(version.trim().into());
+    let version = match registry {
+        Some(registry) => PackageVersion::ThirdParty(ThirdPartyVersion {
+            version: version.trim().into(),
+            registry: registry.into(),
+        }),
```

Comment:
If the file contains `--index-url https://pypi.org/simple`, this would mistakenly identify the package as `ThirdParty`. But maybe that's too obscure to worry about.